include/fuse.h

```diff
@@ -1004,6 +1004,17 @@ int fuse_getgroups(int size, gid_t list[]);
 int fuse_interrupted(void);
 
 /**
+ * Invalidates cache for the given path.
+ *
+ * This calls fuse_lowlevel_notify_inval_inode internally.
+ * This only does something for nodes which are present in
+ * the hashes; it is a no-op for paths to forgotten nodes.
+ *
+ * @return 0 on success, negative error value otherwise.
```

Comment:
Why would you keep possibly useful information from the caller? Especially when there is basically zero cost to do so?

It's not an error. It's an indication that there was no cached value to invalidate. The docs should clearly indicate that. If there are true error conditions that should be pointed out as well.

IMO the high level API already hides too much.